app/models/amazon_product_search.rb

```diff
@@ -15,10 +14,6 @@ def get_matching
     end
   end
 
-  def all_product_data
-    service.get_info(upcs_list_to_string)
-  end
-
   def unique_product_data
     result = upcs.map do |upc|
       all_product_data.find do |product_data|
```

Comment:
Here, every time we call all_product_data, we were firing off another `get_info` call through the Amazon service.